File: Algorithms/Ad Hoc/Oleg and Shares.py
N, K= tuple(map(int, input().split()))
nums = list(map(int, input().split()))
m = min(nums)
print(-1 if [0 for x in nums] != [(x-m) % K for x in nums] else sum([(x-m) // K for x in nums]), end='')

File: Algorithms/Ad Hoc/Oleg and Shares.cpp
#include <cstdio>

using namespace std;

#define MAXN 100000 + 10
#define INF 1<<30

int nums[MAXN];

int main(){
    int N, K;
    scanf("%d %d", &N, &K);
    int min = INF;
    for (int i = 0; i < N; i++) {
       scanf("%d", nums + i);
       if (nums[i] < min) min = nums[i];
    }
    long long sum = 0;
    bool corr = true;
    for (int i = 0; i < N; i++){
        if ((nums[i] - min) % K == 0) sum += (nums[i] - min) / K;
        else {corr = false; break;}
    }
    cout <<  (corr ? sum : -1);
}